components/alibi-explain-server/alibiexplainer/explainer.py

```diff
@@ -0,0 +1,165 @@
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#
+# Originally copied from https://github.com/kubeflow/kfserving/blob/master/python/alibiexplainer/alibiexplainer/explainer.py
+# and since modifed
+#
+
+import json
+import logging
+from enum import Enum
+from typing import List, Any, Mapping, Union, Dict
+import numpy as np
+from alibiexplainer.anchor_images import AnchorImages
+from alibiexplainer.anchor_tabular import AnchorTabular
+from alibiexplainer.anchor_text import AnchorText
+from alibiexplainer.kernel_shap import KernelShap
+from alibiexplainer.integrated_gradients import IntegratedGradients
+from alibiexplainer.explainer_wrapper import ExplainerWrapper
+from alibiexplainer.proto import prediction_pb2
+from alibiexplainer.proto import prediction_pb2_grpc
+from alibi.api.interfaces import Explanation
+import grpc
+import tensorflow as tf
+import alibiexplainer.seldon_http as seldon
+import requests
+import os
+from alibiexplainer.model import ExplainerModel
+from tensorflow import keras
+
+SELDON_LOGLEVEL = os.environ.get('SELDON_LOGLEVEL', 'INFO').upper()
+logging.basicConfig(level=SELDON_LOGLEVEL)
+GRPC_MAX_MSG_LEN = 1000000000
+KFSERVING_PREDICTOR_URL_FORMAT = "http://{0}/v1/models/{1}:predict"
+SELDON_PREDICTOR_URL_FORMAT = "http://{0}/api/v0.1/predictions"
+
+class Protocol(Enum):
+    tensorflow_http = "tensorflow.http"
+    seldon_http = "seldon.http"
+    seldon_grpc = "seldon.grpc"
+
+    def __str__(self):
+        return self.value
+
+class ExplainerMethod(Enum):
+    anchor_tabular = "AnchorTabular"
+    anchor_images = "AnchorImages"
+    anchor_text = "AnchorText"
+    kernel_shap = "KernelShap"
+    integrated_gradients = "IntegratedGradients"
+
+    def __str__(self):
+        return self.value
+
+
+class AlibiExplainer(ExplainerModel):
+    def __init__(self,
+                 name: str,
+                 predictor_host: str,
+                 method: ExplainerMethod,
+                 config: Mapping,
+                 explainer: object = None,
+                 protocol: Protocol = Protocol.seldon_grpc,
+                 tf_data_type: str = None,
+                 keras_model: keras.Model = None ):
+        super().__init__(name)
+        self.predictor_host = predictor_host
+        logging.info("Predict URL set to %s", self.predictor_host)
+        self.method = method
+        self.protocol = protocol
+        self.tf_data_type = tf_data_type
+        logging.info("Protocol is %s",str(self.protocol))
+
+        if self.method is ExplainerMethod.anchor_tabular:
+            self.wrapper:ExplainerWrapper = AnchorTabular(self._predict_fn, explainer, **config)
```

Comment:
Adds extra ExplainerWrapper type only on this one

Reply:
That is just to get mypy to not fail type errors by giving the first reference a type.

Reply:
Ah fair, may be worth adding a comment so it's not removed